spec.md

```diff
@@ -179,6 +212,32 @@ physically present in the final circuit. Refer to the description of the
 instance statement for details on how to instantiate a module
 ([@sec:instances]).
 
+### Public Modules
+
+A public module is a module that is marked with the `public`{.firrtl} keyword.
+This means that this is a module that is usable external to this circuit.  A
+public module thereby has a defined ABI---knowing the FIRRTL representation is
+sufficient to know its compiled representation.
+
+A public module has a number of restrictions:
+
+1. A public module may not be instantiated in the current circuit.
+1. A public module may have no ports of uninferred width.
+1. A public module may have no ports of abstract reset type.
```

Comment:
FWIW also no input probes, can cannot RWProbe a public module's ports.  This is in the text elsewhere already, and may not need to be repeated here.